www_src/pages/map/map.jsx

```diff
@@ -1,66 +1,170 @@
 var React = require('react');
-var render = require('../../lib/render.jsx');
 var Binding = require('../../lib/binding.jsx');
-var Hammer = require('react-hammerjs');
-var Grid = require('./grid.jsx');
+var render = require('../../lib/render.jsx');
+var Link = require('../../components/link/link.jsx');
+var Cartesian = require('../../lib/cartesian');
+var assign = require('react/lib/Object.assign');
+var classNames = require('classnames');
+var elements = require('./fakeElements');
+var {Menu, Primary, Secondary} = require('../../components/action-menu/action-menu.jsx');
 
-var App = React.createClass({
-  mixins: [Binding],
+var Map = React.createClass({
   getInitialState: function () {
     return {
-      zoomLevel: 1,
-      fullyZoomedIn: false
+      selectedEl: '',
+      camera: {
+        x: 0,
+        y: 0
+      },
+      zoom: 0.5
+    };
+  },
+  mixins: [Binding],
+  componentWillMount: function () {
+    var width = 300;
+    var height = 380;
+    var gutter = 20;
+
+    this.cartesian = new Cartesian({
+      allCoords: elements.map(el => el.coords),
+      width,
+      height,
+      gutter
+    });
+  },
+  componentDidMount: function () {
+    var el = this.getDOMNode();
+    var bounding = this.refs.bounding;
+    var boundingEl = bounding.getDOMNode();
+    var startX, startY, deltaX, deltaY;
+    var didMove = false;
+
+    // Set up initial translation
+    this.setState({
```

Comment:
hm. i suppose we could if we create `this.cartesian` in getInitialstate actually